cpu/cc2538/periph/uart.c

```diff
@@ -308,11 +173,72 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
 
 void uart_poweron(uart_t uart)
 {
-    (void) uart;
+    assert(uart < UART_NUMOF);
 
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 }
 
 void uart_poweroff(uart_t uart)
 {
-    (void) uart;
+    assert(uart < UART_NUMOF);
+
+    /* Wait for the TX FIFO to clear */
+    while (uart_config[uart].dev->cc2538_uart_fr.FRbits.TXFF) {}
+
+    uart_config[uart].dev->cc2538_uart_ctl.CTLbits.UARTEN = 0;
 }
+
+void uart_clockon(uart_t uart)
```

Comment:
What are these two functions (clockon/clockoff) about? They are not part of the UART interface, so they should not be non-static. Further they are not used anywhere in the code, so why have them? I suspect that the code from this functions should probalby rather be integrated into the poweron/off functions, right?!